.github/workflows/cdk-e2e.yaml

```diff
@@ -0,0 +1,91 @@
+# Workflow to run cdk integration tests with finch
+name: AWS CDK CI
+
+on:
+  workflow_dispatch:
+  # Run every day at 12am
+  schedule:
```

Comment:
why is this a schedule rule instead of running on PRs and merges? do the tests take too long? it feels like this workflow would be too easy to ignore

Reply:
The reason why it sits in finch-core and also why it does not run against every PR is that we don't want to block finch PRs if cdk adds a feature to it's main branch which is not yet compatible with finch. finch will try to bridge the gap asap but  cdk will also not block it's release if finch is not compatible with some container constructs.  I do agree that it's easy to ignore this workflow. How about adding an action in this workflow to open an issue if the test fails, would that help get better visibility when there is incompatibility?

Reply:
maybe the right move would be to pin cdk & tests to a specific version/commit? that way we can isolate finch regressions.

> How about adding an action in this workflow to open an issue if the test fails, would that help get better visibility when there is incompatibility?

yeah, that's a good idea

Reply:
> maybe the right move would be to pin cdk & tests to a specific version/commit? that way we can isolate finch regressions.

That's an interesting idea to catch regressions in finch but not for cdk. If cdk adds an integration test in their suite or adds another construct that uses docker/finch, we won't be testing that if we pin the commit. 

> How about adding an action in this workflow to open an issue if the test fails, would that help get better visibility when there is incompatibility?

>> yeah, that's a good idea

Cool, I'll add an action to open issues if the test suite fails.

Reply:
> >maybe the right move would be to pin cdk & tests to a specific version/commit? that way we can isolate finch regressions.
>
> That's an interesting idea to catch regressions in finch but not for cdk. If cdk adds an integration test in their suite or adds another construct that uses docker/finch, we won't be testing that if we pin the commit.

we can always bump cdk to a new version/commit. it'd be a pain to do that manually all the time, but maybe we could add a schedule action to open a new PR to bump cdk to a new version and we can test it while merging the PR?

Reply:
Updated to pin the sha. I am going to create another PR to open an issue if the run fails based on this action: https://github.com/marketplace/actions/create-an-issue . Having trouble with the token atm but should be easy to fix.

Reply:
I don't think it makes sense to pin the SHA if we're sticking with the scheduled approach -- that suggestion was meant for a workflow that runs on every Finch PR and blocks merges. Under that approach, limiting the tests to only the Finch changes is helpful because we don't want development on Finch to be blocked by an unrelated change to the CDK.

If we're going with the schedule approach, since it doesn't block Finch PRs, I think it makes sense to always run it against the latest version of CDK to discover issues that changes to the CDK might introduce.

Reply:
Reverted the change back to track main branch.